client/app/scripts/liveblog-edit/module.js

```diff
@@ -167,19 +167,52 @@ define([
         });
     }
 
-    BlogSettingsController.$inject = ['blog', 'api'];
-    function BlogSettingsController(blog, api) {
+    BlogSettingsController.$inject = ['$scope', 'blog', 'api', 'blogService', '$location', 'notify'];
+    function BlogSettingsController($scope, blog, api, blogService, $location, notify) {
+        // set view's model
         var vm = this;
-        // set scope
         angular.extend(vm, {
             blog: blog,
-            blogSettings: {},
-            languages: []
+            blogPreferences: angular.copy(blog.blog_preferences),
+            availableLanguages: [],
+            availableThemes: [],
+            isSaved: true,
+            save: function() {
+                // save on backend and clsoe
+                notify.info(gettext('saving blog settings'));
+                blogService.save(vm.blog._id, {blog_preferences: vm.blogPreferences}).then(function(blog) {
+                    vm.isSaved = true;
+                    vm.blog = blog;
+                    notify.pop();
+                    notify.info(gettext('blog settings saved'));
+                    vm.close();
+                });
+            },
+            reset: function() {
+                // reset vm.blogPreferences's values with the ones from global_preferences (backend)
+                api('global_preferences').query().then(function(global_preferences) {
```

Comment:
`api` is a service but just a service to easeful work with api,
that doesn't exclude using other service to work across controller's functionality.
And using a common controller for something with services where build is not a good ideea.

ex: $http service and having a service that will fetch users, that is used in more than in one place.
 api('global_preferences') is used more then once.